pipeline/lib/storage.ml

```diff
@@ -135,3 +135,33 @@ let mark_closed_pull_requests ~open_pulls (db : Postgresql.connection) =
 
 let mark_closed_pull_requests ~open_pulls ~conninfo =
   Db_util.with_db ~conninfo (mark_closed_pull_requests ~open_pulls)
+
+let get_main_branch_metrics ~repository ~worker ~docker_image
+    (db : Postgresql.connection) =
+  let repo_id = Db_util.string @@ Repository.info repository
+  and commit = Db_util.string @@ Repository.commit_hash repository
+  and worker = Db_util.string @@ worker
+  and docker_image = Db_util.string @@ docker_image in
+  let main_commit =
+    Fmt.str
+      {|(SELECT commit FROM benchmarks
+       WHERE pull_number is NULL AND repo_id=%s AND worker=%s AND docker_image=%s
+       ORDER BY -EXTRACT(epoch FROM run_at) LIMIT 1)
```

Comment:
Nitpick: would `ORDER BY run_at DESC` work? In any case well done, this query is great to handle the multiple entries associated with a single commit!

Reply:
Yes, that works too. Thanks!